nuxeo-web-ui/Jenkinsfiles/server-web-ui.groovy

```diff
@@ -0,0 +1,318 @@
+        container('ftests') {
+          script {
+            echo """
+            ----------------------------------------
+            Deploy Docker Image
+            ----------------------------------------"""
+            // Kubernetes namespace, requires lower case alphanumeric characters
+            def testNamespace = "${CURRENT_NAMESPACE}-server-build-web-ui-release-${NUXEO_MAJOR_VERSION.toLowerCase()}"
+
+            // Create namespace
+            echo 'Create test namespace'
+            sh "kubectl create namespace ${testNamespace}"
+
+            try {
+              echo 'Copy image pull secret to test namespace'
+              sh "kubectl --namespace=platform get secret kubernetes-docker-cfg -ojsonpath='{.data.\\.dockerconfigjson}' | base64 --decode > /tmp/config.json"
+              sh """kubectl create secret generic kubernetes-docker-cfg \
+                --namespace=${testNamespace} \
+                --from-file=.dockerconfigjson=/tmp/config.json \
+                --type=kubernetes.io/dockerconfigjson --dry-run -o yaml | kubectl apply -f -"""
+
+
+              echo 'Deploy nuxeo and external services'
+              try {
+                helmfileTemplate("${testNamespace}", "${TEST_ENVIRONMENT}", 'target')
+                archiveHelmTemplates()
+                helmfileSync("${testNamespace}", "${TEST_ENVIRONMENT}")
+              } catch (e) {
+                sh """
+                  kubectl --namespace=${testNamespace} get event --sort-by .lastTimestamp
+                  kubectl --namespace=${testNamespace} get all,configmaps,secrets
+                  kubectl --namespace=${testNamespace} describe pod --selector=app=${TEST_HELM_RELEASE}
+                  kubectl --namespace=${testNamespace} logs --selector=app=${TEST_HELM_RELEASE}
+                """
+                throw e
+              }
+
+              echo """
+              ------------------------------------------
+              Run Functional Tests
+              ------------------------------------------
+              Nuxeo image version: ${NUXEO_IMAGE_VERSION}
+              Web UI package version: ${NUXEO_WEB_UI_VERSION}
+              Web UI Branch: ${WEB_UI_BRANCH}
+              Node Version: ${NODE_VERSION}
+              Google Chrome Version: ${GOOGLE_CHROME_VERSION}
+              """
+
+              dir(NUXEO_WEB_UI_CLONE){
+                def nuxeoServerUrl = "http://nuxeo.${testNamespace}.svc.cluster.local/nuxeo"
+                sh """ 
+                  npm install
+                  npm run ftest -- --nuxeoUrl=${nuxeoServerUrl}
+                """
+              }
+              setGitHubBuildStatus('docker/deploy', 'Deploy docker image', 'SUCCESS')
+            } catch (e) {
+              archiveScreenshots()
+              setGitHubBuildStatus('docker/deploy', 'Deploy docker image', 'FAILURE')
+              throw e
+            } finally {
+              try {
+                cucumber buildStatus: 'FAILURE',
```

Comment:
Shouldn't the buildStatus depend on the actual result of `npm run ftest`?

Reply:
Yes, the buildStatus depends on the report which in turn is set by the results of `npm run ftest`. 
`Type: String
buildStatus (optional)
Build [result](http://javadoc.jenkins-ci.org/hudson/model/Result.html) to which the build should be set when the report is marked as failed or unstable.`
(from https://www.jenkins.io/doc/pipeline/steps/cucumber-reports/)